crates/ra_syntax/src/grammar/patterns.rs

```diff
@@ -48,11 +48,16 @@ fn atom_pat(p: &mut Parser, recovery_set: TokenSet) -> Option<CompletedMarker> {
     // test literal_pattern
     // fn main() {
     //     match () {
+    //         -1 => (),
     //         92 => (),
     //         'c' => (),
     //         "hello" => (),
     //     }
     // }
+    if p.at(MINUS) {
+        p.bump();
+    }
```

Comment:
This will parse `let -_ = ()` which I think we don't want to allow. I think we need to parse a literal and return from this branch.